lib/scrivener/paginater/ecto/query.ex

```diff
@@ -34,18 +34,31 @@ defimpl Scrivener.Paginater, for: Ecto.Query do
       |> apply(:__schema__, [:primary_key])
       |> hd
 
-    total_entries =
+    query =
       query
       |> exclude(:order_by)
       |> exclude(:preload)
       |> exclude(:select)
       |> exclude(:group_by)
-      |> select([m], count(field(m, ^primary_key), :distinct))
-      |> repo.one
+
+    total_entries =
+      if query.distinct do
+        query
+        |> subquery
+        |> count(repo, primary_key)
```

Comment:
This looks much better. Can we just do this in all cases rather than only doing it for queries with a `distinct` clause? I think using a subquery for counting will solve a lot of the problems we've run into here.